### DISCUSSION TITLE
How to denoise the EBSD data like this.

### DISCUSSION THREAD

**Discussion by yuehengzhang95** [mtex-toolbox/mtex#1424] [Ask Anything]
Hi, I have EBSD data like the figure below
![441651697718_ pic](https://user-images.githubusercontent.com/59850513/166838442-d1326fb8-86f4-4e3c-a8b5-125641135399.jpg).

For some reason (maybe because my crystals are not flat, there are so many noise caused by shadows), my data looks not good. 
I followed the tutorial to generate the EBSD map (like below), but I do not know how to remove the noise which can not be crystals.
I attached my codes here, could you please help me figure this out? Many thanks!

![531651697880_ pic](https://user-images.githubusercontent.com/59850513/166838711-4acd3476-f926-485c-80cd-d59976344438.jpg)

`%% Import Script for EBSD Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('432', [4.1 4.1 4.1], 'mineral', 'Gold', 'color', [0.56 0.74 0.56]),...
  crystalSymmetry('4/mmm', [3.9 3.9 3.9], 'mineral', 'Strontium Titanium Oxide', 'color', [0.53 0.81 0.98])};

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','intoPlane');

%% Specify File Names

% path to files
pname = '/Users/yuehengzhang/Desktop/CMU/LauePUP422/04-18-22';

% which files to be imported
fname = [pname '/Scan10.ang'];

%% Import the Data

% create an EBSD variable containing the data
% ebsd = EBSD.load(fname,CS,'interface','ang',...
%   'convertEuler2SpatialReferenceFrame');

ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame','setting 2');

ebsd = ebsd('Gold');

% reconstruct the grain structure
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'angle',10*degree);

% remove some very small grains
ebsd(grains(grains.grainSize<5)) = [];

% redo grain segementation
% [grains,ebsd.grainId] = calcGrains(ebsd,'angle',10*degree);
% ebsd(grains(grains.grainSize<5)) = [];
% smooth grain boundaries
% grains = smooth(grains,3);

% plot the orientation map
ipfKey = ipfColorKey(ebsd.CS.properGroup);
plot(ebsd,ipfKey.orientation2color(ebsd.orientations))`

**Comment by kilir** [comment]
Hi,
I do not see why there are leftovers. What does `plot(grains('g'),grains('g').grainSize)` (assuming it's gold, otherwise use the perovskite) look like?
Cheers,
Rüdiger
